packages/autocomplete-js/src/autocomplete.ts

```diff
@@ -30,6 +31,7 @@ export function autocomplete<TItem extends BaseItem>(
   const { runEffect, cleanupEffects, runEffects } = createEffectWrapper();
   const { reactive, runReactives } = createReactiveWrapper();
 
+  const id = options.id ?? generateAutocompleteId();
```

Comment:
We need to get it from `props.value.core.id`, otherwise we'll never be able to update the `id` with the Update API.

Reply:
Fixed in https://github.com/algolia/autocomplete/pull/489/commits/5c0eca412700da825c207d4fcc9f9f71dbcfa82f

Does this approach make sense to you?

Reply:
No, we shouldn't need to treat it differently from other options. I meant directly using `props.value.core.id`. Doesn't this work?

Reply:
No, because the id is not a reactive and doesn't seems to be updated and initializing it as a reactive or directly inside `createAutocomplete` will trigger `generateAutocompleteId()`

Reply:
Fixed here: https://github.com/algolia/autocomplete/pull/489/commits/38bc0a7942f72d977c30eca10c7760c8ebb3c217